src/emqx_ws_connection.erl

```diff
@@ -253,7 +253,33 @@ websocket_init([Req, Opts]) ->
                    #{src_address := SrcAddr, src_port := SrcPort} ->
                        {SrcAddr, SrcPort};
                    _ ->
-                       cowboy_req:peer(Req)
+                       {PeerAddr, PeerPort} = cowboy_req:peer(Req),
+                        Addr = case proplists:get_value(proxy_address_header, Opts) of
+                            x_forwarded_for ->
```

Comment:
should get back a string, `HeaderName` then call 
`cowboy_req:header(lowercase(HeaderName), Req, <<>>)`